pkgs/top-level/lua-packages.nix

```diff
@@ -5,140 +5,123 @@
   for each package in a separate file: the call to the function would
   be almost as must code as the function itself. */
 
-{ fetchurl
-, stdenv
+{ stdenv
 , lua
-, pkg-config
-, fetchFromGitHub
-, which
-, writeText
 , pkgs
 , lib
-}@args:
+}:
+
+self:
 
 let
-  packages = (self:
-
-    let
-      callPackage = pkgs.newScope self;
```

Comment:
@Artturin this previously made a scoped `callPackage` available, which was useful for defining lua `packageOverrides`. Was dropping it intentional...?

Reply:
`packageOverrides` are broken after this PR, @Artturin any idea how to resolve this ?

```
    luajit =
      let
        packageOverrides = self: super: rec {
          box = super.callPackage ./pkgs/lua-modules/box { inherit exc };
          exc = super.callPackage ./pkgs/lua-modules/exc { };
          ...
        };
      in
      super.luajit.override {
        inherit packageOverrides;
        self = luajit;
      };
```

error:
```
error: attribute 'callPackage' missing

       at overlays.nix:

          203|         packageOverrides = self: super: rec {
          204|           box = super.callPackage ./pkgs/lua-modules/box { inherit exc; };
             |                 ^
(use '--show-trace' to show detailed location information)
```

Reply:
change `super.` to `self.`

Reply:
thx @Artturin , it works now